packages/discovery-react-components/src/utils/document/transformEnrichment.ts

```diff
@@ -145,23 +143,18 @@ function getAllAttributesInRelation({ relations }: Relations): Attributes[] {
 
 // Function checks if attribute is valid and has location data in it.
 // Since have observed noisy data before so this function is required.
-function isValidAttribute(attr: Attributes): boolean {
-  if (attr && attr.location && attr.location.begin && attr.location.end) {
-    return true;
-  }
-  return false;
+function hasLocationData(attr: Attributes): boolean {
+  return !!(attr && attr.location && attr.location.begin && attr.location.end);
 }
 
 export default function transformEnrichment(enrichedHtml: EnrichedHtml[]): EnrichedHtml[] {
-  const newEnrichedHtml = { ...enrichedHtml };
-
-  if (newEnrichedHtml && newEnrichedHtml[0]) {
-    const enrichmentName = getEnrichmentName(newEnrichedHtml[0]);
-    const enrichment = newEnrichedHtml[0][enrichmentName];
+  if (enrichedHtml && enrichedHtml[0]) {
+    const enrichmentName = getEnrichmentName(enrichedHtml[0]);
+    const enrichment = enrichedHtml[0][enrichmentName];
     if (enrichment) {
       modelMapping[enrichmentName](enrichment, enrichmentName);
     }
   }
 
-  return newEnrichedHtml;
+  return enrichedHtml;
```

Comment:
It's not evident what is happening here. It seems that `modelMapping` is updating something inside of `enrichment`, which is a part of `enrichedHtml`. So that would be considered a function that has side-effects, resulting in not easily readable/maintainable code.

I think a better approach would be to make the `modelMapping` functions **pure**: they take in some parameters and return a value, but do not in any way change anything with the incoming parameters.

Then, in this function, we would have something like:

```javascript
const updatedEnrichment = modelMapping[enrichmentName](enrichment, enrichmentName);
enrichedHtml[0][enrichmentName] = updatedEnrichment;
```

That way, as someone who is reading this function, I can clearly see that the function `modelMapping` is returning some value, which is then stored within `enrichedHtml`.